dis_snek/models/discord/guild.py

```diff
@@ -1230,13 +1230,19 @@ async def fetch_channel(self, channel_id: Snowflake_Type) -> Optional["models.TY
 
         """
         channel_id = to_snowflake(channel_id)
-        if channel_id in self._channel_ids:
+        if channel_id in self._channel_ids or not self._client.gateway_started:
+            # The latter check here is to see if the bot is running with the gateway.
+            # If not, then we need to check the API since only the gateway
+            # populates the channel IDs
+
             # theoretically, this could get any channel the client can see,
             # but to make it less confusing to new programmers,
             # i intentionally check that the guild contains the channel first
             try:
-                return await self._client.fetch_channel(channel_id)
-            except NotFound:
+                channel = await self._client.fetch_channel(channel_id)
+                if channel.guild.id == self.id:
```

Comment:
`channel._guild_id` (or guild_id not sure)

Reply:
Changed, forgot about that property